crates/api/src/site/purge/person.rs

```diff
@@ -29,17 +29,29 @@ pub async fn purge_person(
   // Read the person to get their images
   let person_id = data.person_id;
 
-  let local_user = LocalUserView::read_person(&mut context.pool(), person_id).await?;
-  let pictrs_uploads =
-    ImageUpload::get_all_by_local_user_id(&mut context.pool(), &local_user.local_user.id).await?;
+  if let Ok(local_user) = LocalUserView::read_person(&mut context.pool(), person_id).await {
```

Comment:
This prevented purging of remote users, same as https://github.com/LemmyNet/lemmy/pull/4171. Need to be careful when reading LocalUser unconditionally.